### PR TITLE
Make LatLngBounds obey its argument order

### LINKED ISSUE #2415: Allow "fitBounds" to span the date line
**mapbox-gl-js version**:

### Steps to Trigger Behavior

 1. Use fitBounds with correct SW/NE argument ordering and expect that a dateline-crossing bounds is accepted

### Expected Behavior

The map is centered over the dateline over a certain polygon.

### Actual Behavior

The fitBounds constructor calls LatLngBounds convert, which claims to support a specific argument order but really doesn't differentiate if the arguments are flipped ( see https://github.com/mapbox/mapbox-gl-js/pull/2414 for a PR). Even when the arguments are treated as strict, the fitBounds algorithm also doesn't handle the case of a dateline-crossing extent.

### PR DESCRIPTION
**Pull request by tmcw**
fixes #2415

### CODE REVIEW

**Comment by lucaswoj** [comment]
Could you provide a little more context for this PR? What do you mean by "obey its argument order"? Can you demonstrate the behavior with a unit test?

**Comment by tmcw** [comment]
See the referenced issue #2415 for the details

File: js/geo/lng_lat_bounds.js [hunk 1]
```diff
@@ -36,6 +36,42 @@ LngLatBounds.prototype = {
...
+    setNE: function(obj) {
+        var ne = this._ne, ne2;
+
+        if (obj instanceof LngLat) {
+            ne2 = obj;
+        } else {
+            return obj ? this.setNE(LngLat.convert(obj) || LngLatBounds.convert(obj)) : this;
+        }
+        this._ne = new LngLat(ne2.lng, ne2.lat);
+        return this;
```
Comment (lucaswoj) [inline js/geo/lng_lat_bounds.js]: Could this code be rewritten as

```js
setNE: function(ne) {
    ne = LngLat.convert(ne) || LngLatBounds.convert(ne) || this;
    this._ne = new LngLat(ne.lng, ne.lat);
    return this;
}
```

**Comment by lucaswoj** [comment]
Refactored and fixed failing unit tests. 

There are some unit test "expected value" updates that raise a red flag in my head.

@mourner @morganherlocker What do yout think about these changes? How painful will this breaking change be to our users?

**Comment by mourner** [comment]
@lucaswoj not sure how many people used the reverse order for the bounds, but it's easy to upgrade at least.

**Comment by lucaswoj** [comment]
@mourner Sounds good to me. Is this ready to 🚢?

**Pull request merged**